contracts/connection-router/src/state.rs

```diff
@@ -421,6 +421,12 @@ impl Display for ChainName {
     }
 }
 
+impl PartialEq<String> for ChainName {
+    fn eq(&self, other: &String) -> bool {
+        self.0 == other.to_lowercase()
```

Comment:
I am confused here. Do we want case-sensitive or case-insensitive? This is technically neither.

Reply:
trying to do case insensitive, chain name is always in lower case `ChainName(s.to_lowercase())`